zebra/kernel_netlink.c

```diff
@@ -311,18 +322,30 @@ static int netlink_socket(struct nlsock *nl, unsigned long groups,
 		snl.nl_family = AF_NETLINK;
 		snl.nl_groups = groups;
 
+		if (ext_group_size) {
```

Comment:
(non-blocking) for a bit better style - the `if (ext_group_size)` is only useful for the case where `SOL_NETLINK` is not defined, i.e. the second error message; otherwise it's just a zero-sized loop.  If you move down the `if (ext_group_size)` you can drop a level of indentation here, make things more readable :smile:

Reply:
ext_group_size only works with SOL_NETLINK support